expr/consolidations/consolidations.go

```diff
@@ -375,19 +381,30 @@ func AggCount(v []float64) float64 {
 	return float64(n)
 }
 
-// AggCount counts non-NaN points
 func AggDiff(v []float64) float64 {
-	res := v[0]
-	if len(v) == 1 {
-		return res
-	}
-	for _, vv := range v[1:] {
+	safeValues := make([]float64, 0)
+	for _, vv := range v {
 		if !math.IsNaN(vv) {
-			res -= vv
+			safeValues = append(safeValues, vv)
 		}
 	}
 
-	return res
+	if len(safeValues) > 0 {
+		res := safeValues[0]
+		if len(safeValues) == 1 {
+			return res
+		}
+
+		for _, vv := range safeValues[1:] {
+			if !math.IsNaN(vv) {
```

Comment:
Check really needed ?

Reply:
May be simplified with one-pass scan and boolean flags is all is NaN ? It's more simple and has better performance (no copy slice elements).